src/System.Buffers.Experimental/System/Buffers/BufferExtensions.cs

```diff
@@ -16,8 +16,11 @@ public static void Pipe(this Transformation transformation, ReadOnlyBytes source
         {
             int afterMergeSlice = 0;
 
+            // TODO: "ReadOnlySpan<byte> remainder = stackalloc byte[0]" would fit better here, 
+            //       but it emits substandard IL, see https://github.com/dotnet/roslyn/issues/21952
+            //
             // make 'remainder' formally stack-referring or we won't be able to reference stack data later
-            ReadOnlySpan<byte> remainder = stackalloc byte[0]; 
+            var remainder = true ? new ReadOnlySpan<byte>() : stackalloc byte[0];
```

Comment:
nit: change from var to ReadOnlySpan<byte>, esp since in the future, the r.h.s would look like `stackalloc byte[0];`

Reply:
Will do. I was not sure if mentioning the type twice would be too verbose.